testsuite/shared/src/main/java/org/jboss/as/test/config/ContainerConfig.java

```diff
@@ -0,0 +1,270 @@
+     *   <li>Environment variable: TESTSUITE_CONFIG_PROPERTIES=/path/to/testsuite-config.properties</li>
+     * </ul>
+     * If no file is specified, an empty Properties object is returned.
+     *
+     * @return the loaded properties, or an empty Properties object if no file is configured
+     * @throws RuntimeException if the properties file cannot be loaded
+     */
+    private static Properties getPropertiesFromFile() {
+        if (properties == null) {
+            properties = new Properties();
+            // Properties file location can be passed as e.g. -Dtestsuite.config.properties=/some-path/testsuite-config.properties
+            String configFileLocation = System.getProperty(PROPERTIES_FILE_PROPERTY_NAME) != null ?
+                    System.getProperty(PROPERTIES_FILE_PROPERTY_NAME) :
+                    System.getenv(PROPERTIES_FILE_PROPERTY_NAME.replace(".", "_").toUpperCase());
+            if (!Strings.isNullOrEmpty(configFileLocation)) {
+                logger.fine(String.format("Loading config properties from file: %s", configFileLocation));
+                try (FileReader reader = new FileReader(convertToFile(configFileLocation))) {
+                    properties.load(reader);
+                } catch (IOException e) {
+                    throw new RuntimeException(String.format("Could not load config properties from %s", configFileLocation), e);
+                }
+            }
+        }
+        return properties;
+    }
+
+    /**
+     * Converts a file location string to a File object.
+     * <p>
+     * Attempts to parse the location as a URI first. If that fails, treats it as a file system path.
+     *
+     * @param configFileLocation the file location as a string (URI or file path)
+     * @return the File object representing the location
+     */
+    private static File convertToFile(String configFileLocation) {
+        try {
+            return new File(new URI(configFileLocation));
+        } catch (URISyntaxException | IllegalArgumentException e) {
+            return Paths.get(configFileLocation).toFile();
+        }
+    }
+
+    /**
+     * Retrieves a configuration property value from multiple sources.
+     * <p>
+     * Checks sources in the following order of precedence:
+     * <ol>
+     *   <li>System property (e.g., -Dkey=value)</li>
+     *   <li>Environment variable (key converted to uppercase with dots replaced by underscores)</li>
+     *   <li>Properties file (if configured)</li>
+     *   <li>Default value (if provided)</li>
+     * </ol>
+     *
+     * @param key the property key
+     * @param defaultValue the default value to use if the property is not found
+     * @return the property value, or the default value if not found
+     */
+    private static String getProperty(String key, String defaultValue) {
+        String value = Strings.isNullOrEmpty(System.getProperty(key)) ?
+                System.getenv(key.replace(".", "_").toUpperCase())
```

Comment:
This should be `toUpperCase(Locale.ROOT)`. Maybe that is not a big deal for the test suite, but it potentially could be.